llvm/test/CodeGen/NVPTX/global-offset-multiple-entry-points.ll

```diff
@@ -0,0 +1,108 @@
+; RUN: opt -globaloffset %s -S -o - | FileCheck %s
+; ModuleID = 'multiple-entry-points.bc'
+source_filename = "multiple-entry-points.ll"
+target datalayout = "e-i64:64-i128:128-v16:16-v32:32-n16:32:64"
+target triple = "nvptx64-nvidia-nvcl-sycldevice"
+
+; This test checks that the pass works with multiple entry points.
```

Comment:
Can we add a check with an entry point being called from another entry point?

Reply:
As mentioned in https://github.com/intel/llvm/pull/1773#discussion_r435204261 it is assumed that entry points don't call other entry points. Additionally we have an assertion in `getEntryPointMetadata` that requires an entry point to not have uses.

Reply:
> As mentioned in [#1773 (comment)](https://github.com/intel/llvm/pull/1773#discussion_r435204261) it is assumed that entry points don't call other entry points.

I thought that calling entry point from device code is allowed in SYCL. Are you saying it's not support on NV?

Reply:
How would you achieve this ? The entry point is generated by the frontend, the user has no way to refer to it. Do I miss understand you comment ?

Reply:
Sorry, disregard previous comment. I replied without thinking.

You mean that at NV target is using OpenCL kernels generated by SYCL FE. Right?
If so, than you are right and current implementation protect us from this use case and probably assertion is enough.

Reply:
That is not a kernel entry point from the OpenCL or CUDA API, just a regular function.

Reply:
What is "That"?

Are `_ZTS12first_kernel` and `_ZTS13second_kernel` defined by user or auto-generated by the compiler?
If they are user functions I assume they can be called from an entry point.

Reply:
> You mean that at NV target is using OpenCL kernels generated by SYCL FE. Right?

Yes